packages/light.js/src/rpc/utils/createRpc.ts

```diff
@@ -79,26 +42,39 @@ const createRpc = <Source, Out>(metadata: Metadata<Source, Out>) => (
     if (metadata.pipes && isFunction(metadata.pipes)) {
       pipes.push(...metadata.pipes(api));
     }
-    pipes.push(multicast(() => subject$), refCount());
-    if (withoutLoading === true) {
-      pipes.push(withoutLoadingOperator());
-    }
-    pipes.push(distinctValues());
+    pipes.push(multicast(() => subject$), refCount(), distinctValues());
 
-    // Add a field in the calledWithArgs object, so that we know this function has
-    // been called with these particular args in the app. See overview.js on
-    // how this is used.
-    if (!metadata.calledWithArgs) {
-      metadata.calledWithArgs = {};
+    return source$.pipe(...pipes) as Observable<Out>;
+  },
+  {
+    length: false, // Dynamic args length
+    normalizer: (args: any) => {
+      // Custom memoization function, i.e. create an unique id from the args.
+      // `args` is arguments object as accessible in memoized function
+      return `${args[0].name}${args[1].provider.id}${JSON.stringify(
+        Array.from(args).slice(2)
+      )}`;
     }
-    metadata.calledWithArgs[prune(args)] = subject$;
-
-    return source$.pipe(...pipes);
-  };
+  }
+);
 
-  Object.assign(rpc$, frequencyMixins, { metadata });
+/**
+ * Add metadata to an RpcObservable, and transform it into a ReplaySubject(1).
+ * It's a currified function.
+ *
+ * @ignore
+ * @param metadata - The metadata to add.
+ * @example
+ * createRpc(metadata)(options) returns a RpcObservable.
+ * createRpc(metadata)(options)(someArgs) returns an Observable.
```

Comment:
I think the syntax is now the other way around? `createRpc(metadata)(someArgs)(options)`

Reply:
I put the options first everywhere in the latest commit